File: ImperatorToCK3/Source/Mappers/DeathReasonMapper/DeathReasonMapping.cpp
#include "DeathReasonMapping.h"
#include "ParserHelpers.h"
#include "CommonRegexes.h"



mappers::DeathReasonMapping::DeathReasonMapping(std::istream& theStream) {
	registerKeyword("ck3", [this](std::istream& theStream) {
		ck3Reason = commonItems::getString(theStream);
	});
	registerKeyword("imp", [this](std::istream& theStream) {
		impReasons.emplace(commonItems::getString(theStream));
	});
	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);

	parseStream(theStream);
	clearRegisteredKeywords();
}

File: ImperatorToCK3/Mappers/DeathReason/DeathReasonMapping.cs
using commonItems;
using System.Collections.Generic;

namespace ImperatorToCK3.Mappers.DeathReason;

public sealed class DeathReasonMapping {
	public SortedSet<string> ImperatorReasons { get; } = new();
	public string? Ck3Reason { get; private set; }

	public DeathReasonMapping(BufferedReader mappingReader) {
		var parser = new Parser();
		parser.RegisterKeyword("ck3", reader => Ck3Reason = reader.GetString());
		parser.RegisterKeyword("ir", reader => ImperatorReasons.Add(reader.GetString()));
		parser.RegisterRegex(CommonRegexes.Catchall, ParserHelpers.IgnoreAndLogItem);

		parser.ParseStream(mappingReader);
	}
}